mmdet/apis/train.py

```diff
@@ -68,6 +68,51 @@ def set_random_seed(seed, deterministic=False):
         torch.backends.cudnn.benchmark = False
 
 
+def scale_lr(cfg, logger):
+    """Automatically scaling LR according to GPU number and sample per GPU.
+
+    Args:
+        cfg (config): training config.
+        logger (logging.Logger): logger.
+    """
+    enable_auto_scale_lr = cfg.get('enable_auto_scale_lr', False)
+    if enable_auto_scale_lr is False:
+        logger.info(f'You disabled automatically scaling LR.')
+        return
+
+    # default batch size of pre-train model
+    original_batch_size = cfg.get('default_batch_size', None)
+    assert original_batch_size is not None
+
+    # calculate the batch size
+    gpu_number = len(cfg.gpu_ids)
+    batch_size = gpu_number * cfg.data.samples_per_gpu
+
+    logger.info(f'You are using {gpu_number} GPU(s) '
+                f'and {cfg.data.samples_per_gpu} samples per gpu '
+                f'means the batch is {batch_size}.')
+
+    if batch_size != original_batch_size:
+        # Get original LR from config
+        original_lr = cfg.optimizer.get("lr", 0)
+        assert original_lr != 0
+
+        # scale LR according to paper
+        # [linear scaling rule](https://arxiv.org/abs/1706.02677)
+        scaled_lr = (batch_size / original_batch_size) * original_lr
+        cfg.optimizer.update({"lr": scaled_lr})
+
+        logger.info(f'While default batch size of this model '
+                    f'is {original_batch_size}, '
+                    f'automatically scaling LR '
+                    f'from {original_lr} to {scaled_lr}')
+
+    else:
+        logger.info(f'Which match the '
```

Comment:
We should remove f here because there is no variable.

Reply:
Done